hathitrust/ht_full_text_search
# dockerizing and documenting the search application

**Experimental PR (Work in Progress)**

This is a temporal repository that contains an application that reproduces the Solr queries we make in the application (Advance search). In next sprints, this repository should disappear and its logic will be integrate at ht_indexer repository. 

**What we could do with this application?**
* Run one Solr query or a batch of queries using different Solr versions (Solr 6 and Solr 8).
* Save the query results in a CSV file to manually analyze the results
* Compare CSV files generated with different Solr engines or with any variation of Solr parameter.
* A batch of queries were tested to make sure the current code obtains the same results we showed in the platform
* The main structure was extracted from perl and transform into python. 
* Some functions are commented in the code because they come from perl and as this application is a WIP I prefer to keep it then until decide if we need it or not.

**How could you test this PR?**

- In your work directory, run the commands below. You will start up a Solr server with a sample of data with 150 documents.

        `docker build -t full-text-searcher .`
        `docker-compose up -d`

- Use the following command to run a query 
        `compose exec full_text_searcher python ht_full_text_search/ht_full_text_searcher.py --env dev --query_string "natural history" --operator None --query_config ocronly`

This query should return a total of 26 documents. If you scroll up the terminal results you should see the same results that in the image below.

![image](https://github.com/hathitrust/ht_full_text_search/assets/47088146/b341672b-afa8-4065-a5b5-e5686fcc6d32)

**Use case we could consider in a near future**:
* Create a class that manages the Solr output to be able to generate logs of Solr results (Useful for production. It could support some of Ryan's task)
* Run the application in Kubernetes to easy query production server
